source/utilities.tex

```diff
@@ -4442,6 +4451,551 @@
 \tcode{*this} does not contain a value.
 \end{itemdescr}
 
+\rSec2[optional.optional.ref]{Partial specialization of \tcode{optional} for reference types}
+
+\rSec3[optional.optional.ref.general]{General}
+\begin{codeblock}
+namespace std {
+  template<class T>
+  class optional<T&> {
+    public:
+      using value_type     = T;
+      using iterator       = @\impdefnc@;            // see~\ref{optional.ref.iterators}
+
+    public:
+      // \ref{optional.ref.ctor}, constructors
+      constexpr optional() noexcept = default;
+      constexpr optional(nullopt_t) noexcept : optional() {}
+      constexpr optional(const optional& rhs) noexcept = default;
+
+      template<class Arg>
+        constexpr explicit optional(in_place_t, Arg&& arg);
+      template<class U>
+        constexpr explicit(@\seebelow@) optional(U&& u) noexcept(@\seebelow@);
+      template<class U>
+        constexpr explicit(@\seebelow@) optional(optional<U>& rhs) noexcept(@\seebelow@);
+      template<class U>
+        constexpr explicit(@\seebelow@) optional(const optional<U>& rhs) noexcept(@\seebelow@);
+      template<class U>
+        constexpr explicit(@\seebelow@) optional(optional<U>&& rhs) noexcept(@\seebelow@);
+      template<class U>
+        constexpr explicit(@\seebelow@) optional(const optional<U>&& rhs) noexcept(@\seebelow@);
```

Comment:
For optional<T> the ctors are in this order:

```
constexpr optional() noexcept;
constexpr optional(nullopt_t) noexcept;
constexpr optional(const optional&);
constexpr optional(optional&&) noexcept(see below );
template<class... Args>
  constexpr explicit optional(in_place_t, Args&&...);
template<class U, class... Args>
  constexpr explicit optional(in_place_t, initializer_list<U>, Args&&...);
template<class U = remove_cv_t<T>>
  constexpr explicit(see below ) optional(U&&);
template<class U>
  constexpr explicit(see below ) optional(const optional<U>&);
template<class U>
  constexpr explicit(see below ) optional(optional<U>&&);
```

Is there any reason to deviate from that here? I don't recall discussing this in LWG.

Reply:
Obviously some of those ctors aren't present for `optional<T&>` but for the ones that are, I think it makes sense to be consistent.

Reply:
Oh, I suppose they already are consistent, just with additional ones inserted. OK, ignore me, sorry for the noise.